icu4c/source/test/cintltst/unumberformattertst.c

```diff
@@ -212,6 +216,64 @@ static void TestExampleCode() {
 }
 
 
+static void TestSimpleNumberFormatter() {
+    // This is the example in usimplenumberformatter.h
+    UErrorCode ec = U_ZERO_ERROR;
+    USimpleNumberFormatter* uformatter = usnumf_openForLocale("bn", &ec);
+    USimpleNumber* unumber = usnum_openForInt64(1000007, &ec);
+    UFormattedNumber* uresult = unumf_openResult(&ec);
+    usnumf_format(uformatter, unumber, uresult, &ec);
+    int32_t len;
+    const UChar* str = ufmtval_getString(unumf_resultAsValue(uresult, &ec), &len, &ec);
+    if (assertSuccess("Formatting end-to-end 1", &ec)) {
+        assertUEquals("Should produce a result in Bangla digits", u"১০,০০,০০৭", str);
+    }
+
+    usnumf_close(uformatter);
+    uformatter = usnumf_openForLocaleAndGroupingStrategy("de-CH", UNUM_GROUPING_ON_ALIGNED, &ec);
+
+    usnumf_formatInt64(uformatter, 4321, uresult, &ec);
+    str = ufmtval_getString(unumf_resultAsValue(uresult, &ec), &len, &ec);
+    if (assertSuccess("Formatting end-to-end 2", &ec)) {
+        assertUEquals("Should produce a result with Swiss symbols", u"4’321", str);
+    }
+
+    usnum_setToInt64(unumber, 98765, &ec);
+    usnum_multiplyByPowerOfTen(unumber, -2, &ec);
+    usnum_roundTo(unumber, -1, UNUM_ROUND_HALFDOWN, &ec);
+    usnum_setMinimumIntegerDigits(unumber, 4, &ec);
+    usnum_setMinimumFractionDigits(unumber, 3, &ec);
+    usnum_truncateStart(unumber, 1, &ec);
+    usnum_setSign(unumber, UNUM_SIMPLE_NUMBER_PLUS_SIGN, &ec);
+
+    usnumf_format(uformatter, unumber, uresult, &ec);
+    str = ufmtval_getString(unumf_resultAsValue(uresult, &ec), &len, &ec);
+    if (assertSuccess("Formatting end-to-end 3", &ec)) {
+        assertUEquals("Should produce a result with mutated number", u"+0’007.600", str);
+    }
+
+    // Cleanup:
+    unumf_closeResult(uresult);
+    usnum_close(unumber);
+    usnumf_close(uformatter);
+}
+
+
+/**
+ * Formats an integer using this SimpleNumberFormatter.
+ *
+ * For more control over the formatting, use USimpleNumber.
+ *
+ * @draft ICU 73
+ */
+U_CAPI void U_EXPORT2
+usnumf_formatInt64(
+    const USimpleNumberFormatter* uformatter,
+    int64_t value,
+    UFormattedNumber* uresult,
+    UErrorCode* ec);
```

Comment:
why is this here? redundant with the public header?

Reply:
Oops, removed